main.cpp

```diff
@@ -62,11 +64,25 @@ std::unique_ptr<GenericUI> CreateUI(int argc, char** argv) {
   return std::make_unique<CommandLineUI>();
 }
 
-}  // namespace
+std::unique_ptr<GenericUI> CreateUI(int argc, char** argv) {
+  return std::make_unique<WindowsUI>();
+}
+
+const bool Type_of_UI = ReadCommandLine(argc, argv);
```

Comment:
This function ReadCommandLine is not defined. What we want here is something like:

if user runs the program with --text, create the CommandLineUI. If user starts with --gui, instantiate your class instead. Otherwise quit (or pick one of the UI by default).